### PR TITLE
fix problem in dump and add log

### PR DESCRIPTION
**Pull request by hutuxian**
### PR types

Bug fixes
### PR changes

Others
### Describe

* Fix the field length in LoD scenario
* Fix the missed lod info when copy tensor in dump field
* Add some log to make debug easy

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_boxps.py [hunk 1]
```diff
@@ -161,8 +161,13 @@ def create_dataset():
                 sync_steps=-1)
             optimizer.minimize(loss)
 
-            program._pipeline_opt[
-                "dump_fields"] = ["fc.tmp_0", "fc.tmp_0@GRAD", "hehe"]
+            program._pipeline_opt["dump_fields"] = [
+                "fc.tmp_0", "fc.tmp_0@GRAD", "hehe", "z", "reduce_mean_3.tmp_0"
+            ]
+            # hehe: not in scope
```
Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_boxps.py]: what's this for?

Comment (gavin1332) [reply python/paddle/fluid/tests/unittests/test_boxps.py]: rename hehe

**Review by gavin1332 — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_boxps.py (hunk 1 above)
Comment (hutuxian) [reply python/paddle/fluid/tests/unittests/test_boxps.py]: done

**Pull request merged**